mods/aero-flyout-fix.wh.cpp

```diff
@@ -0,0 +1,453 @@
+// ==WindhawkMod==
+// @id              aero-flyout-fix
+// @name            Aero Flyout Fix
+// @description     Applies thick borders to legacy system flyouts and make them float
+// @version         1.0.0
+// @author          aubymori
+// @github          https://github.com/aubymori
+// @include         explorer.exe
+// @include         SndVol.exe
+// @architecture    x86-64
+// @compilerOptions -lgdi32 -lshell32
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# Aero Flyout Fix
+Since Windows 10, many of the legacy tray flyouts either don't have thick borders or don't
+float away from the taskbar. This mod adds back thick borders and makes flyouts float where
+applicable.
+
+# IMPORTANT: READ!
+Windhawk needs to hook into `winlogon.exe` to successfully capture Explorer starting. Please
+navigate to Windhawk's Settings, Advanced settings, More advanced settings, and make sure that
+`winlogon.exe` is in the Process inclusion list.
+
+**Before**:
+
+![Before](https://raw.githubusercontent.com/aubymori/images/main/aero-flyout-fix-before.png)
+
+**After**:
+
+![After](https://raw.githubusercontent.com/aubymori/images/main/aero-flyout-fix-after.png)
+*/
+// ==/WindhawkModReadme==
+
+#include <windhawk_utils.h>
+#include <winerror.h>
+
+#define RECTWIDTH(rect)  ((rect).right - (rect).left)
+#define RECTHEIGHT(rect) ((rect).bottom - (rect).top)
+
+const UINT FLYOUT_OFFSET = 8;
+
+/* Adjust a window's position to be pushed away from the taskbar */
+POINT AdjustWindowPosForTaskbar(HWND hWnd)
+{
+    HMONITOR hm = MonitorFromWindow(hWnd, MONITOR_DEFAULTTONEAREST);
+    HDC hDC = GetDC(hWnd);
+    int offset = MulDiv(FLYOUT_OFFSET, GetDeviceCaps(hDC, LOGPIXELSY), 96);
+
+    RECT rc;
+    GetWindowRect(hWnd, &rc);
+
+    MONITORINFO mi = { sizeof(MONITORINFO) };
+    GetMonitorInfoW(hm, &mi);
+
+    int dx = 0, dy = 0;
+    PLONG plrc = (PLONG)&rc;
+    PLONG plwrc = (PLONG)&mi.rcWork;
+    for (int i = 0; i < 4; i++)
+    {
+        int curOffset = plwrc[i] - plrc[i];
+        curOffset = (curOffset < 0) ? -curOffset : curOffset;
+
+        if (curOffset < offset)
+        {
+            int *set = (i % 2 == 0) ? &dx : &dy;
+            if (i > 1)
+            {
+                *set -= offset - curOffset;
+            }
+            else
+            {
+                *set += offset - curOffset;
+            }
+        }
+    }
+    return { rc.left + dx, rc.top + dy };
+}
+
+/* == EXPLORER.EXE HOOKS == */
```

Comment:
Tip: Are you familiar with `#pragma region`?
You can wrap code with something like:
```cpp
#pragma region explorer_hooks
// ...
#pragma endregion  // explorer_hooks
```
Which is a more conventional way of marking a block of code, and has the benefit of being collapsible in the editor.